host_core/lib/host_core/host.ex

```diff
@@ -73,13 +73,102 @@ defmodule HostCore.Host do
 
     publish_host_started(labels, friendly_name)
 
-    {:ok,
-     %State{
-       host_key: opts.host_key,
-       lattice_prefix: opts.lattice_prefix,
-       friendly_name: friendly_name,
-       labels: labels
-     }}
+    state = %State{
+      host_key: opts.host_key,
+      lattice_prefix: opts.lattice_prefix,
+      friendly_name: friendly_name,
+      labels: labels
+    }
+
+    if config_service_enabled?(opts) do
+      {:ok, state, {:continue, :load_supp_config}}
+    else
+      {:ok, state}
+    end
+  end
+
+  # truthy
+  defp config_service_enabled?(opts) do
+    String.upcase(opts.config_service_enabled) in [
+      "TRUE",
+      "YES",
+      "Y",
+      "YOU BETCHA",
+      "YUPPERS",
+      "TOTES",
+      "ENABLED"
+    ]
+  end
+
+  @impl true
+  def handle_continue(:load_supp_config, state) do
+    topic = "wasmbus.cfg.#{state.lattice_prefix}"
+    Logger.debug("Requesting supplemental host configuration via topic '#{topic}'")
+
+    state =
+      with {:ok, supp_config} <-
+             HostCore.ConfigServiceClient.request_configuration(
+               state.labels,
+               topic
+             ) do
+        %State{state | supplemental_config: supp_config}
+      else
+        {:error, e} ->
+          Logger.error("Failed to obtain supplemental configuration: #{inspect(e)}")
+          state
+      end
+
+    {:noreply, state, {:continue, :process_supp_config}}
+  end
+
+  @impl true
+  def handle_continue(:process_supp_config, %State{supplemental_config: nil} = state) do
+    {:noreply, state}
+  end
+
+  # NOTE - we do this work in a handle continue because we need the registry credentials
+  # to be a part of this process's state prior to attempting to start the components in the
+  # supp config's prestarts
+  @impl true
+  def handle_continue(:process_supp_config, %State{supplemental_config: sc} = state) do
+    aps = Map.get(sc, "autoStartProviders", [])
+    as = Map.get(sc, "autoStartActors", [])
+
+    Logger.info(
+      "Processing supplemental configuration: #{length(aps)} providers, #{length(as)} actors"
+    )
+
+    Task.start(fn ->
+      aps
+      |> Enum.each(fn prov ->
+        if !Map.has_key?(prov, "imageReference") || !Map.has_key?(prov, "linkName") do
+          Logger.error("Not enough information on auto-start provider configuration. Bypassing.")
+        else
+          if String.starts_with?(prov["imageReference"], "bindle") do
```

Comment:
to prevent a wild edge case where someone has their OCI server named `bindlebasket.azurecr.io`, we should probably check for starting with `bindle://`